proposals/nnnn-distributed-actors.md

```diff
@@ -0,0 +1,1621 @@
+
+This allows us to pass actors as references across distributed boundaries:
+
+```swift
+distributed actor Person {
+  distributed func greet(_ greeting: String) async throws {
+    log.info("I was greeted: '\(greeting)' yay!")
+  }
+}
+
+distributed actor Greeter { 
+  var greeting: String = "Hello, there!"
+  
+  distributed func greet(person: Person) async throws {
+    try await person.greet("\(greeting)!")
+  }
+}
+```
+
+#### Discovering Existing Instances
+
+As discussed, identities are crucial to locate and resolve an opaque identity into a real and "live" actor reference that we can send messages through.
+
+However, how do we communicate an identity of one actor from one node to another if to identify _any distributed actor_ actor we need to know their identity to begin with? This immediately ends up in a "catch-22" situation, where in order to share an identity with another actor, we need to know _that_ actor's identity, but we can't know it, since we were not able to communicate with it yet!
+
+Luckily, this situation is not uncommon and has established solutions that generally speaking are forms of _service discovery_. Swift already offers a general purpose service discovery library with [swift-service-discovery](https://github.com/apple/swift-service-discovery), however its focus is very generic and all about services, which means that for a given well-known service name e.g. "HelloCluster", we're able to lookup which _nodes_ are part of it, and therefore we should attempt connecting to them. Implementations of service discovery could be using DNS, specific key-value stores, or kubernetes APIs to locate pods within a cluster. 
+
+This is great, and solves the issue of locating _nodes_ of a cluster, however we also need to be able to locate specific _distributed actors_, that e.g. implement some specific protocol. For example, we would like to, regardless of their location locate all `Greeter` actors in our distributed actor system. We can use the well-known type name Greeter as key in the lookup, or we could additionally qualify it with some identifier for example only to find those `Greeter` actors which use the language `pl` (for Polish).
+
+In practice, _how_ such lookups are performed is highly tied to the underlying transport, and thus the transport library should provide a specific pattern to perform those lookups. We call this pattern the `Receptionist`, and it may take the following protocol shape:
+
+```swift
+protocol Receptionist: DistributedActor { 
+    @discardableResult
+    public func register<Guest>(
+        _ guest: Guest,
+        with key: Reception.Key<Guest>,
+        replyTo: ActorRef<>? = nil
```

Comment:
This could do with some de-Javafication but the intent is clear.

Reply:
Ah, good catch the reply to is not necessary, fixed :)